tox.ini

```diff
@@ -1,12 +1,20 @@
 [tox]
-envlist=check,py{26,27,34,35,36,py}-pytest{28,29,30}
+envlist=check,py{26,27,34,35,36,py}
 
 [testenv]
 commands=py.test {posargs:testing/}
 deps=
-  pytest28: pytest~=2.8.0
-  pytest29: pytest~=2.9.0
-  pytest30: pytest~=3.0.0
+  pytest
+
+[testenv:pytest-master]
+commands=py.test {posargs:testing/}
+deps=
+  git+https://github.com/pytest-dev/pytest.git@master
+
+[testenv:pytest-features]
+commands=py.test {posargs:testing/}
```

Comment:
those could just be factors as well

Reply:
Those are not factors on purpose because I don't think we need to test pytest master and develop on all python versions again, only on py27 and py36 seems good enough to me to save CI time. What do you think?

Reply:
we dont have to express those factors for all python versions

Reply:
I agree, and that's what I am doing here... perhaps I'm misunderstanding something, could you clarify.

For reference, take a look at the build matrix that was executed by this change: https://travis-ci.org/pytest-dev/pluggy/builds/270596643

Reply:
you made own environments, not factors

Reply:
Sorry @RonnyPfannschmidt, still don't know what you want me to do. 😕 

Let me try to illustrate.

Right now we have the following builds:

```
py26  (pytest: 3.2.1)
py27  (pytest: 3.2.1) 
py34  (pytest: 3.2.1) 
py34  (pytest: 3.2.1) 
py35  (pytest: 3.2.1)
py36  (pytest: 3.2.1)
pypy  (pytest: 3.2.1)
py37  (pytest: 3.2.1)
benchmark (python: 2.7)
benchmark (python: 3.6)
check (python: 2.7)
check (python: 3.6)
docs (python: 3.6)
pytest-master  (python: 2.7, pytest: master)
pytest-master  (python: 3.6, pytest: master)
pytest-features  (python: 2.7, pytest: features)
pytest-features  (python: 3.6, pytest: features)
```

> you made own environments, not factors

From this I understand you want me to change `[envlist]` to:

```ini
[tox]
envlist=check,py{26,27,34,35,36,py}-pytest{release,master,features}
```

But this would explode the builds matrix to:

```
py26  (pytest: 3.2.1)
py27  (pytest: 3.2.1) 
py34  (pytest: 3.2.1) 
py34  (pytest: 3.2.1) 
py35  (pytest: 3.2.1)
py36  (pytest: 3.2.1)
pypy  (pytest: 3.2.1)
py37  (pytest: 3.2.1)
py26  (pytest: master)
py27  (pytest: master) 
py34  (pytest: master) 
py34  (pytest: master) 
py35  (pytest: master)
py36  (pytest: master)
pypy  (pytest: master)
py37  (pytest: master)
py26  (pytest: features)
py27  (pytest: features) 
py34  (pytest: features) 
py34  (pytest: features) 
py35  (pytest: features)
py36  (pytest: features)
pypy  (pytest: features)
py37  (pytest: features)
benchmark (python: 2.7)
benchmark (python: 3.6)
check (python: 2.7)
check (python: 3.6)
docs (python: 3.6)
```

Is that it? If so I don't think we need it.

If that's not it, could you clarify then? If you know exactly what syntax you want feel free to write it as well.

Reply:
```envlist=check,py{26,27,34,35,36,py}-pytestrelease,py{27,36}-pytest{master,features}```

Reply:
OK, I think I got it now, thanks.